### ISSUE TITLE
🐛 [firebase_auth] FirebaseAuth.instance.currentUser is null

### ISSUE THREAD

**Issue by xinoxapps** [FirebaseExtended/flutterfire#6019]
## Bug report

**Describe the bug**

Since updating from 1.0.1 to 1.1.2 the call to FirebaseAuth.instance.currentUser is always null unless you add a delay.

### Steps to reproduce

await Firebase.initializeApp();
var user = FirebaseAuth.instance.currentUser;
// is always null
print("currentUser: $user");

await Future.delayed(Duration(milliseconds: 1), () {
  var user = FirebaseAuth.instance.currentUser;
  // user is not null
  print("currentUser after delay: $user");
});

1. Login
2. Restart the app.
3. Active user should be present

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.12.3
Flutter SDK 2.0.6
workspaces 1.0.14+24

dependencies:
- bloc 7.0.0 [meta]
- cached_network_image 3.0.0 [flutter flutter_cache_manager octo_image]
- carousel_slider 4.0.0-nullsafety.0 [flutter]
- cloud_firestore 1.0.7 [cloud_firestore_platform_interface cloud_firestore_web firebase_core firebase_core_platform_interface flutter meta]
- cloud_functions 1.0.5 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- cupertino_icons 1.0.3
- custom_sliding_segmented_control 0.1.1 [flutter]
- equatable 2.0.0 [collection meta]
- firebase_analytics 8.0.2 [firebase_analytics_platform_interface firebase_analytics_web firebase_core flutter meta]
- firebase_auth 1.1.2 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.1.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_dynamic_links 2.0.2 [firebase_core flutter]
- firebase_storage 8.0.5 [firebase_core firebase_core_platform_interface firebase_storage_platform_interface firebase_storage_web flutter]
- fl_chart 0.20.1 [flutter path_drawing equatable pedantic]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_bloc 7.0.0 [flutter bloc provider]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]
- flutter_svg 0.22.0 [flutter meta path_drawing vector_math xml]
- google_fonts 2.0.0 [flutter http path_provider crypto]
- image_picker 0.7.4 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface image_picker_for_web]
- intl 0.17.0 [clock path]
- json_annotation 4.0.1
- package_info 2.0.0 [flutter]
- path_provider 2.0.1 [flutter path_provider_platform_interface path_provider_macos path_provider_linux path_provider_windows]
- permission_handler 7.1.0 [flutter meta permission_handler_platform_interface]
- webview_flutter 2.0.4 [flutter]

transitive dependencies:
- archive 3.1.2 [crypto path]
- async 2.6.1 [meta collection]
- characters 1.1.0
- charcode 1.2.0
- clock 1.1.0
- cloud_firestore_platform_interface 4.0.3 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 1.0.7 [cloud_firestore_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- cloud_functions_platform_interface 5.0.3 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.0.5 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.15.0
- crypto 3.0.1 [collection typed_data]
- ffi 1.0.0
- file 6.1.0 [meta path]
- firebase 9.0.1 [http http_parser js]
- firebase_analytics_platform_interface 2.0.0 [flutter meta]
- firebase_analytics_web 0.3.0 [firebase firebase_analytics_platform_interface flutter flutter_web_plugins meta]
- firebase_auth_platform_interface 4.2.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 1.1.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core_platform_interface 4.0.0 [flutter meta plugin_platform_interface]
- firebase_core_web 1.0.2 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- firebase_storage_platform_interface 2.0.3 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_storage_web 1.0.5 [async firebase_core firebase_core_web firebase_storage_platform_interface flutter flutter_web_plugins http js meta]
- flutter_blurhash 0.6.0 [flutter meta pedantic]
- flutter_cache_manager 3.0.1 [clock collection file flutter http image path path_provider pedantic rxdart sqflite uuid]
- flutter_plugin_android_lifecycle 2.0.1 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- http 0.13.3 [async http_parser meta path pedantic]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.0.2 [archive meta xml]
- image_picker_for_web 2.0.0 [image_picker_platform_interface meta flutter flutter_web_plugins]
- image_picker_platform_interface 2.1.0 [flutter meta http plugin_platform_interface]
- js 0.6.3
- meta 1.3.0
- nested 1.0.0 [flutter]
- octo_image 1.0.0+1 [flutter flutter_blurhash]
- path 1.8.0
- path_drawing 0.5.0 [vector_math meta path_parsing flutter]
- path_parsing 0.2.0 [vector_math meta]
- path_provider_linux 2.0.0 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 2.0.0 [flutter]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.1 [path_provider_platform_interface meta path flutter ffi win32]
- pedantic 1.11.0
- permission_handler_platform_interface 3.3.0 [flutter meta plugin_platform_interface]
- petitparser 4.1.0 [meta]
- platform 3.0.0
- plugin_platform_interface 2.0.0 [meta]
- process 4.2.1 [file path platform]
- provider 5.0.0 [collection flutter nested]
- rxdart 0.26.0
- sky_engine 0.0.99
- source_span 1.8.1 [collection path term_glyph]
- sqflite 2.0.0+3 [flutter sqflite_common path]
- sqflite_common 2.0.0+2 [synchronized path meta]
- string_scanner 1.1.0 [charcode source_span]
- synchronized 3.0.0
- term_glyph 1.2.0
- typed_data 1.3.0 [collection]
- uuid 3.0.4 [crypto]
- vector_math 2.1.0
- win32 2.0.5 [ffi]
- xdg_directories 0.2.0 [meta path process]
- xml 5.1.0 [collection meta petitparser]
```

</details>

---

**Comment by markusaksli-nc** [comment]
Reproducible only on iOS with `firebase_auth: ^1.1.2`
<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 2.3.0-1.0.pre.226, on macOS 11.2.3 20D91 darwin-arm, locale en-GB)
    • Flutter version 2.3.0-1.0.pre.226 at /Users/nevercode/development/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 51ca131e39 (11 hours ago), 2021-05-03 19:54:04 -0700
    • Engine revision 378e4dbc41
    • Dart version 2.14.0 (build 2.14.0-48.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Users/nevercode/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/nevercode/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/nevercode/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/201.7199119/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      � https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] Connected device (3 available)
    • Nevercode’s iPhone (mobile) • b668e524315069f3db3661ac11ff1f66afafebdb • ios            • iOS 14.5
    • macOS (desktop)             • macos                                    • darwin-arm64   • macOS 11.2.3 20D91 darwin-arm
    • Chrome (web)                • chrome                                   • web-javascript • Google Chrome 90.0.4430.93

• No issues found!
```
</details>

**Comment by ocastx** [comment]
This seems to be fixed in `firebase_auth: ^1.1.3`

**Comment by Ehesp** [comment]
Yep this was recently fixed, thanks @ocastx
